src/Microsoft.Data.Analysis/IDataView.Extension.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Microsoft.Data.Analysis
+{
+    public static class IDataViewExtensions
+    {
+        public static DataFrame ToDataFrame(this IDataView dataView, long maxRows = -1)
+        {
+            return ToDataFrame(dataView, maxRows, null);
+        }
+
+        public static DataFrame ToDataFrame(this IDataView dataView, params string[] selectColumns)
+        {
+            return ToDataFrame(dataView, -1, selectColumns);
+        }
+
+        public static DataFrame ToDataFrame(this IDataView dataView, long maxRows, params string[] selectColumns)
+        {
+            DataViewSchema schema = dataView.Schema;
+            List<DataFrameColumn> columns = new List<DataFrameColumn>(schema.Count);
+
+            HashSet<string> selectColumnsSet = null;
+            if (selectColumns != null && selectColumns.Length > 0)
+            {
+                selectColumnsSet = new HashSet<string>(selectColumns);
+            }
+
+            List<DataViewSchema.Column> activeColumns = new List<DataViewSchema.Column>();
+            foreach (DataViewSchema.Column column in schema)
+            {
+                long length = maxRows >= 0 ? maxRows : long.MaxValue;
+                length = Math.Min(length, dataView.GetRowCount() ?? 0);
```

Comment:
I'm worried about this line. When `dataView.GetRowCount()` returns null it just means that the row count is not known yet. Here its being set to 0 and since the Min is being picked it will pick 0. Reading the rest of the code for the `DataFrameColumns` it seems that if the count is 0 then it pulls in no data. Is that correct? If so, we may want to handle this differently because null is returned fairly often from my experience.

Reply:
> it seems that if the count is 0 then it pulls in no data. Is that correct?

Not quite. `length` just initializes (and allocates memory) the columns to that size with a default value. It doesn't pull in any data in the constructor. 

> dataView.GetRowCount() returns null it just means that the row count is not known yet.

QQ for you here: Let's say dataView.GetRowCount returns null. If I used `dataView.GetRowCursor` now, will it still enumerate the rows? (even though the row count is not known) . If that is the case, then we can get rid of specifying the length of the `DataFrameColumn`s upfront and just append values to the columns as we enumerate the IDataView

Reply:
Yeah, it will enumerate the rows. Getting the count doesn't affect the cursor at all. Typically it only knows the row count after you have enumerated all the rows anyways.